routes/era5wrf.py

```diff
@@ -0,0 +1,559 @@
+        end_date (datetime): User-provided end date
+        coverage_meta (dict): ERA5-WRF coverage metadata
+
+    Returns:
+        True if valid, or Flask response tuple for errors
+    """
+    try:
+        # get actual temporal bounds from coverage description
+        min_date, max_date = get_era5_temporal_bounds(coverage_meta)
+
+        # validate for chronological order
+        if start_date > end_date:
+            return (
+                render_template(
+                    "422/invalid_date_range.html",
+                    start_date=start_date.strftime("%Y-%m-%d"),
+                    end_date=end_date.strftime("%Y-%m-%d"),
+                    min_date=min_date.strftime("%Y-%m-%d"),
+                    max_date=max_date.strftime("%Y-%m-%d"),
+                ),
+                422,
+            )
+
+        # validate for within coverage date range
+        if start_date < min_date or end_date > max_date:
+            return (
+                render_template(
+                    "422/invalid_year.html",
+                    start_year=start_date.year,
+                    end_year=end_date.year,
+                    min_year=min_date.year,
+                    max_year=max_date.year,
+                ),
+                422,
+            )
+
+        return True
+
+    except ValueError as e:
+        # in trouble if we can't parse coverage description metadata
+        return render_template("500/server_error.html"), 500
+
+
+def package_era5wrf_point_data(
+    data_dict, coverage_meta, start_date=None, end_date=None
+):
+    """Package ERA5-WRF data with time-first structure.
+
+    Args:
+        data_dict (dict): Variable names mapped to data arrays from fetch
+        coverage_meta (dict): Coverage metadata containing time axis
+        start_date (datetime, optional): Start date for filtering
+        end_date (datetime, optional): End date for filtering
+
+    Returns:
+        dict: Time-first structured data {date: {variable: value}}
+    """
+
+    time_index = generate_time_index_from_coverage_metadata(coverage_meta)
+
+    # filter time index if date range provided
+    if start_date and end_date:
+        mask = (time_index.date >= start_date.date()) & (
+            time_index.date <= end_date.date()
+        )
+        time_index = time_index[mask]
+
+    # package data with time keys at top level
+    packaged_data = {}
+    for i, timestamp in enumerate(time_index):
+        date_key = timestamp.strftime("%Y-%m-%d")
+        packaged_data[date_key] = {}
+
+        # add each variable's value for this date
+        for variable, values in data_dict.items():
+            if i < len(values) and values[i] is not None:
+                # round to 1 decimal good for current variable set, might need to change later
+                packaged_data[date_key][variable] = round(values[i], 1)
+
+    return packaged_data
+
+
+@routes.route("/era5wrf/")
+@routes.route("/era5wrf/abstract/")
```

Comment:
We might want to make a ticket here to remove the `/abstract` endpoints throughout, that feels very early-days here I think?  (I know: I suggested it!  Not useful anymore?)

Reply:
ticketed in #618